api/resolvers/sub.js

```diff
@@ -315,6 +315,63 @@ export default {
       notifyTerritoryTransfer({ models, sub, to: user })
 
       return updatedSub
+    },
+    unarchiveTerritory: async (parent, { name, hash, hmac, ...data }, { me, models, lnd }) => {
+      if (!me) {
+        throw new GraphQLError('you must be logged in', { extensions: { code: 'UNAUTHENTICATED' } })
+      }
+
+      const oldSub = await models.sub.findUnique({ where: { name } })
+      if (!oldSub) {
+        throw new GraphQLError('sub not found', { extensions: { code: 'BAD_INPUT' } })
+      }
+      if (oldSub.status !== 'STOPPED') {
+        throw new GraphQLError('sub is not archived', { extensions: { code: 'BAD_INPUT' } })
+      }
+      if (oldSub.billingType === 'ONCE') {
+        // sanity check. this should never happen but leaving this comment here
+        // to stop error propagation just in case and document that this should never happen.
+        // #defensivecode
+        throw new GraphQLError('sub should not be archived', { extensions: { code: 'BAD_INPUT' } })
+      }
+
+      let billingCost = TERRITORY_COST_MONTHLY
+      let billPaidUntil = datePivot(new Date(), { months: 1 })
+      if (data.billingType === 'ONCE') {
+        billingCost = TERRITORY_COST_ONCE
+        billPaidUntil = null
+      } else if (data.billingType === 'YEARLY') {
+        billingCost = TERRITORY_COST_YEARLY
+        billPaidUntil = datePivot(new Date(), { years: 1 })
+      }
```

Comment:
`nextBilling` in `lib/territory` should be able to do this for you.

Reply:
Done in 9420f1dc